tfx/components/example_gen/driver_test.py

```diff
@@ -127,6 +129,46 @@ def testResolveExecProperties(self):
           pattern: "span02/version02/split2/*"
         }""", updated_input_config)
 
+    # Test driver behavior using RangeConfig with static range.
+    self._exec_properties[utils.INPUT_CONFIG_KEY] = json_format.MessageToJson(
+        example_gen_pb2.Input(splits=[
+            example_gen_pb2.Input.Split(
+                name='s1',
+                pattern='span{SPAN:2}/version{VERSION}/split1/*'),
+            example_gen_pb2.Input.Split(
+                name='s2',
+                pattern='span{SPAN:2}/version{VERSION}/split2/*')
+        ]),
+        preserving_proto_field_name=True)
+
+    self._exec_properties[utils.RANGE_CONFIG_KEY] = json_format.MessageToJson(
+        range_config_pb2.RangeConfig(
+            static_range=range_config_pb2.StaticRange(start_span_number=1,
+                                                      end_span_number=1)),
+        preserving_proto_field_name=True)
+    self._example_gen_driver.resolve_exec_properties(self._exec_properties,
+                                                     None, None)
+    self.assertEqual(self._exec_properties[utils.SPAN_PROPERTY_NAME], 1)
+    self.assertEqual(self._exec_properties[utils.VERSION_PROPERTY_NAME], 1)
+    self.assertRegex(
+        self._exec_properties[utils.FINGERPRINT_PROPERTY_NAME],
+        r'split:s1,num_files:1,total_bytes:9,xor_checksum:.*,sum_checksum:.*\nsplit:s2,num_files:1,total_bytes:9,xor_checksum:.*,sum_checksum:.*'
+    )
+    updated_input_config = example_gen_pb2.Input()
+    json_format.Parse(self._exec_properties[utils.INPUT_CONFIG_KEY],
+                      updated_input_config)
+    # Check if correct span inside static range is selected.
+    self.assertProtoEquals(
+        """
+        splits {
+          name: "s1"
+          pattern: "span01/version01/split1/*"
```

Comment:
qq, in this case the pattern: span{SPAN:2}/version{VERSION} the version does not have a width config. should the expected string : span01/version1/split1/

is the zero padding some default behavior? or we carry it over from SPAN:2 ?

if it is the former, consider to update docstring / document, if the the latter, this is an incorrect behavior, consider to address these issues in a separate CL.

Reply:
it does not carry over from `{SPAN:2}`.

look at previous usage of the `{SPAN}` spec, before i introduced width modifier (and the version spec), span was retrieved by doing a greedy regex capture (and retrieving the padding as well). this is different from internal behavior i think, since based on proto docstring, internal requires width with any kind of zero padding, while OSS will just try to capture the entire zero padded span number for you.

IIRC, by default, TFX OSS tries to capture the entire span number with zero padding using the greedy regex, so maybe we need to update docstring.